### PR TITLE
Smaller thumbnail size for participants who are not streaming/sharing screen.

### PR DESCRIPTION
**Pull request by StefanDIvanov**
Until now, the width and height of thumbnails in tile view are calculated as follows:
Based on number of participants, a grid is generated, trying to maintain a square.
The new smaller thumbnails, are excluded from this grid.
* Changed the grid of thumbnails to be calculated only based on streaming participants
* Added different style for thumbnails of non streaming participants, which is 15% of the width of the client's viewport and height calculated based on other videos ratio.

### CODE REVIEW

File: modules/UI/videolayout/Filmstrip.js [hunk 1]
```diff
@@ -39,30 +39,51 @@ const Filmstrip = {
...
+        if (thumbs.localThumb) {
+            thumbs.localThumb.css(bigVideoCSS);
         }
 
         $('.avatar-container').css({
             height: `${avatarSize}px`,
             width: `${avatarSize}px`
         });
+
+        Array.from(thumbs.remoteThumbs).forEach(videoThumb => {
```
Comment (venimus) [inline modules/UI/videolayout/Filmstrip.js]: I think you can use spread `[...thumbs.remoteThumbs].`

Comment (StefanDIvanov) [reply modules/UI/videolayout/Filmstrip.js]: updated

File: modules/UI/videolayout/Filmstrip.js [hunk 2]
```diff
@@ -39,30 +39,51 @@ const Filmstrip = {
...
         $('.avatar-container').css({
             height: `${avatarSize}px`,
             width: `${avatarSize}px`
         });
+
+        [...thumbs.remoteThumbs].forEach(videoThumb => {
+            const $thumb = $(videoThumb);
+
+            // Smaller video
+            if ($thumb.hasClass('without-camera')) {
```
Comment (venimus) [inline modules/UI/videolayout/Filmstrip.js]: please don't use any `else`'s, rather use early-returns. It's very important rule (mandatory).
To do that generally put "unhappy paths" to be first (this is checking for errors, not-logged in users etc), in this case there are none. Then follow with "happy paths', where the "default-behavior" usually comes first. The structure should consist of several linear `if (case) { doStuff; return; }`. Also keeping the function body in the 50-lines constraint is the best.
```
// this is a happy-path but handles a default-behavior
if (!$thumb.hasClass..) {
    $thumb.css(bigVideoCSS);
    return;
}
// this is happy-path but rather a main logic (the biggest block)
const ratio = width / height;
...
```

File: modules/UI/videolayout/SmallVideo.js [hunk 3]
```diff
@@ -556,10 +558,39 @@ export default class SmallVideo {
...
+     * Else order > 10
+     * Also adds special className 'without-camera' to apply
+     * different styles (smaller width and height)
+     */
+    reArrangeVideos() {
+        if (!this.isLocal
+            && (this.displayMode === DISPLAY_VIDEO || this.displayMode === DISPLAY_VIDEO_WITH_NAME)) {
+            this.container.style.order = 1;
+            this.$container.removeClass('without-camera');
+        } else if (!this.isLocal
```
Comment (venimus) [inline modules/UI/videolayout/SmallVideo.js]: else is redundant here

File: modules/UI/videolayout/SmallVideo.js [hunk 4]
```diff
@@ -556,10 +558,39 @@ export default class SmallVideo {
...
+            && (this.displayMode === DISPLAY_AVATAR_WITH_NAME || this.displayMode === DISPLAY_AVATAR)) {
+            this.container.style.order = 11;
+            this.$container.addClass('without-camera');
```
Comment (venimus) [inline modules/UI/videolayout/SmallVideo.js]: not important really, but we usually prefer to use `toggleClass('class', true)` and `toggleClass('class', false)` which is easier to maintain later

File: modules/UI/videolayout/Filmstrip.js (hunk 2 above)
Comment (StefanDIvanov) [reply modules/UI/videolayout/Filmstrip.js]: fixed

File: modules/UI/videolayout/SmallVideo.js (hunk 3 above)
Comment (StefanDIvanov) [reply modules/UI/videolayout/SmallVideo.js]: fixed

File: modules/UI/videolayout/SmallVideo.js (hunk 4 above)
Comment (StefanDIvanov) [reply modules/UI/videolayout/SmallVideo.js]: fixed

Comment (venimus) [reply modules/UI/videolayout/SmallVideo.js]: In fact you can really simplify that whole block. Put `order:1` /`order:11` into css, and just toggle 2 classes: 1st that is "bring on stage"(order 1), second that is "bring back" (order: 11) and one for "default" (probably no class, no order)

Comment (venimus) [reply modules/UI/videolayout/SmallVideo.js]: Though it seems acceptable now

Comment (StefanDIvanov) [reply modules/UI/videolayout/SmallVideo.js]: Im wondering why i didn't do it this way initially. Will fix

**Review by venimus — DISMISSED** [review]
lgtm (didn't test)

**Review by dincho — DISMISSED** [review]

**Review by dincho — APPROVED** [review]

**Pull request merged**